cpp/include/cuvs/neighbors/ivf_pq.hpp

```diff
@@ -23,6 +23,13 @@
 
 namespace cuvs::neighbors::ivf_pq {
 
+template <typename IdxT>
```

Comment:
One of the benefits to this design is that the only thing the users should have to know about in the public APIs is the `pimpl` class. That's what decouples all of the internals and allows us to code the headers to a common interface contract. 

Given that- I think we should move the implementations for the impls (and iface), as well as `build_owning` and `build_view` into the source files / detail namespace, because the user shouldn't need to know they exist.